classes/controller/edit_controller.php

```diff
@@ -212,27 +212,35 @@ public function update_post_action() {
             $post       = $DB->get_record('hsuforum_posts', array('id' => $postid), '*', MUST_EXIST);
             $discussion = $DB->get_record('hsuforum_discussions', array('id' => $post->discussion, 'forum' => $forum->id), '*', MUST_EXIST);
 
-            if (empty($groupid)) {
-                $groupid = -1;
+            foreach ($groupids as $gid) {
+                if (empty($groupids)) {
+                    $groupids[] = -1;
+                } else {
```

Comment:
If it is empty, the foreach won't iterate and this code won't execute. Please take another look at this